format/macho/macho.go

```diff
@@ -559,7 +586,71 @@ func machoDecode(d *decode.D, in interface{}) interface{} {
 		}
 		ncmdsIdx++
 	})
-	return nil
+}
+
+func fatParse(d *decode.D) {
+	// Go to start of the file again
+	d.SeekAbs(0)
+	d.FieldStruct("fat_header", func(d *decode.D) {
+		d.FieldRawLen("magic", 8*8)
+		narchs := d.FieldU32("narchs")
+		narchsIdx := 0
+		d.FieldStructArrayLoop("archs", "arch", func() bool {
+			return narchsIdx < int(narchs)
+		}, func(d *decode.D) {
+			// parse FatArch
+			d.FieldStruct("fat_arch", func(d *decode.D) {
+				// beware cputype and cpusubtype changes from ofile header to fat header
+				cpuType := d.FieldU32("cputype", cpuTypes)
+				d.FieldU32("cpusubtype", cpuSubTypes[int64(cpuType)])
+				d.FieldU32("offset")
+				d.FieldU32("size")
+				d.FieldU32("align")
+			})
+		})
+		for i := 0; uint64(i) < narchs; i++ {
+			// parse ofiles
+			ofileDecode(d)
+		}
+	})
+}
+
+func arParse(d *decode.D) {
+	// Go to start of the file again
+	d.SeekAbs(0)
+	d.FieldStruct("ar_header", func(d *decode.D) {
+		name := d.FieldUTF8NullFixedLen("name", 16)
+		d.FieldUTF8NullFixedLen("date", 12)
+		d.FieldUTF8NullFixedLen("uid", 6)
+		d.FieldUTF8NullFixedLen("gid", 6)
+		d.FieldUTF8NullFixedLen("mode", 6)               // parse_octal ?? TODO
```

Comment:
Can do something like https://github.com/wader/fq/blob/master/format/ar/ar.go#L31 .. wonder some more generic sym mapper should be added to the api for this, seems common.

Reply:
Did a PR to factor this out https://github.com/wader/fq/pull/111